### PR TITLE
data capture in app

### PR DESCRIPTION
**Pull request by aitorrf**

### CODE REVIEW

File: library/src/main/java/com/voysis/sevice/local/modules/AudioSaver.kt [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
+            audioOutput?.write(buf)
         }
     }
 
-    private fun init(acousticQuery: AcousticQuery) {
-        id = acousticQuery.id
-        audioOutput = FileOutputStream(File(filesDir, id.plus(".raw"))).channel
+    private fun init() {
+        tempFile = File(filesDir, Date().time.toString().plus(".raw"))
+        audioOutput = FileOutputStream(tempFile).channel
```
Comment (lduffy-voysis) [inline library/src/main/java/com/voysis/sevice/local/modules/AudioSaver.kt]: these two lines could probably be put into the newQuery() method and remove private init()

File: library/src/main/java/com/voysis/sevice/local/modules/AudioSaver.kt [hunk 2]
```diff
@@ -1,47 +1,50 @@
...
+            buf.flip()
+            audioOutput?.write(buf)
         }
     }
 
-    private fun init(acousticQuery: AcousticQuery) {
-        id = acousticQuery.id
-        audioOutput = FileOutputStream(File(filesDir, id.plus(".raw"))).channel
+    private fun init() {
+        tempFile = File(filesDir, Date().time.toString().plus(".raw"))
```
Comment (ttroy50) [inline library/src/main/java/com/voysis/sevice/local/modules/AudioSaver.kt]: Should we allocate some kind of query id and use that as the filename? That way if we also want to save the transcription or any other information later we can have something to link them.

Comment (jcftang) [reply library/src/main/java/com/voysis/sevice/local/modules/AudioSaver.kt]: that would imply having a process to ingest these files and allow for other operations, would the right thing to do is to have a separate tool/process that uploads the data via the rest api if the user chooses to do so?

Comment (aitorrf) [reply library/src/main/java/com/voysis/sevice/local/modules/AudioSaver.kt]: the idea of this implementation is call it from the app, outside of the sdk. I don't have the id until we are in the feedback. If we want add more information later we could add more temporal files and rename all of them. I can change this tempFile for something somantically more correct.

**Pull request merged**